File: api-service/src/main/java/com/govtweets/apiservice/Tweet.java
package com.govtweets.apiservice;

public class Tweet {

    private final long id;
    private final String text;
    private final long authorId;
    private final boolean isRetweet;
    private final boolean isReply;
    private final String creationDate;

    Tweet(long id, String text, long authorId, boolean isRetweet, boolean isReply, String creationDate) {
        this.id = id;
        this.text = text;
        this.authorId = authorId;
        this.isRetweet = isRetweet;
        this.isReply = isReply;
        this.creationDate = creationDate;
    }

    public long getId() {
        return this.id;
    }

    public String getText() {
        return this.text;
    }

    public long getAuthorId() {
        return this.authorId;
    }

    public boolean isRetweet() {
        return this.isRetweet;
    }

    public boolean isReply() {
        return this.isReply;
    }

    public String getCreationDate() {
        return this.creationDate;
    }
}

File: src/api/tweets/Tweet.ts
import {ITweet} from "./ITweet";

export default class Tweet implements ITweet
{
    id: number;
    text: string;
    authorId: number;
    handle: string;
    isRT: boolean;
    isReplyTweet: boolean;
    creationDate: Date;

    constructor(id: number, text: string, authorId: number, handle: string, isRetweet: boolean, isReply: boolean, creationDate: Date)
    {
        this.id = id;
        this.text = text;
        this.authorId = authorId;
        this.handle = handle;
        this.isRT = isRetweet;
        this.isReplyTweet = isReply;
        this.creationDate = creationDate;
    }

    getAuthorId(): number
    {
        return this.authorId;
    }

    getHandle(): string
    {
        return this.handle;
    }

    getText(): string
    {
        return this.text;
    }

    getTweetId(): number
    {
        return this.id;
    }

    isRetweet(): boolean
    {
        return this.isRT;
    }

    isReply(): boolean
    {
        return this.isReplyTweet;
    }

    isOriginalTweet(): boolean
    {
        return !this.isRetweet() && !this.isReply();
    }
}

export type TweetType =
{
    id: number;
    text: string;
    authorId: number;
    handle: string,
    isRT: boolean;
    isReplyTweet: boolean;
    creationDate: string;
}